### PR TITLE
Use safe-buffer

### PR DESCRIPTION
**Pull request by fanatid**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,7 +28,7 @@ function decodeRaw (buffer, version) {
 }
 
 function encodeRaw (version, privateKey, compressed) {
-  var result = new Buffer(compressed ? 34 : 33)
+  var result = Buffer.alloc(compressed ? 34 : 33)
```
Comment (dcousens) [inline index.js]: `allocUnsafe`?

Comment (fanatid) [reply index.js]: We serialize private key to there, I thought that we do not use `allocUnsafe` in such cases.

Comment (dcousens) [reply index.js]: We should throw if `privateKey` is not `Buffer` and is not `.length === 32`

Comment (fanatid) [reply index.js]: Yea.. we need more checks here for nice error messages.

**Pull request merged**